include/boost/url/url.hpp

```diff
@@ -233,14 +233,77 @@ class BOOST_SYMBOL_VISIBLE url
 
     This function attempts to resolve a URL
     reference `ref` against the base URL `base`
-    which must satisfy the <em>absolute-URI</em>
+    in a manner similar to that of a web browser
+    resolving an anchor tag. The base URL
+    must satisfy the <em>absolute-URI</em>
     grammar.
-    This process is defined in detail in
-    rfc3986 (see below).
-    The result of the resolution is placed
-    into `dest`.
+
+    The resolution process works as if the
+    relative reference is appended to the base
+    URI and the result is normalized.
+
+    Relative references are only usable when
+    in the context of a base absolute URI.
+    This process of resolving a relative
+    <em>reference</em> within the context of
+    a <em>base</em> URI is defined in detail
+    in rfc3986 (see below).
+
+    In real applications, the base URI for a
+    reference can be obtained from one of four
+    sources:
+
+    @li Base URI embedded in content
+    @li Base URI of the encapsulating entity
+    @li URI use to retrieve the entity
+    @li Default Base URI (application-dependent)
+
+    For instance, considering HTML documents:
+
+    @li The tag `base` can be used to embed a base
+        URI in the content.
+    @li An anchor fragment uses the encapsulating
+        document as a base URI
+    @li The URL used to retrieve the current page
+        is used as base to other links
+    @li If none of the conditions above apply,
+        any base URL depends the context of the
+        application
```

Comment:
...the hell? I can't understand anything

Reply:
I'll simplify that then